pkg/specgen/generate/kube/kube.go

```diff
@@ -511,45 +515,83 @@ func parseMountPath(mountPath string, readOnly bool, propagationMode *v1.MountPr
 	return dest, opts, nil
 }
 
+func probeToHealthConfig(probe *v1.Probe) (*manifest.Schema2HealthConfig, error) {
+	var commandString string
+	failureCmd := "exit 1"
+	probeHandler := probe.Handler
+
+	// configure healthcheck on the basis of Handler Actions.
+	switch {
+	case probeHandler.Exec != nil:
+		cmd, err := json.Marshal(probeHandler.Exec.Command)
```

Comment:
it's not immediately obvious to me why this works. do you have a go doc reference or code you based this off that could explain why/how it works?

Reply:
https://github.com/containers/podman/blob/02b7866e6000c8f1b5ddbef78bfc008cdecb6ba1/pkg/specgen/generate/kube/kube.go#L562-L580

the function `makeHealthCheck` supports input string as a json array :D